MatrixSDK/Data/Store/MXFileStore/MXFileStore.m

```diff
@@ -496,57 +506,143 @@ - (MXMemoryRoomStore*)getOrCreateRoomStore:(NSString*)roomId
+    if (!backup)
+    {
+        return [storeRoomsPath stringByAppendingPathComponent:roomId];
+    }
+    else
+    {
+        return [self.storeBackupRoomsPath stringByAppendingPathComponent:roomId];
+    }
 }
 
-- (void)checkFolderExistenceForRoom:(NSString*)roomId
+- (NSString*)storeBackupRoomsPath
 {
-    NSString *roomFolder = [self folderForRoom:roomId];
+    NSString *storeBackupRoomsPath;
+
+    if (backupEventStreamToken)
+    {
+        storeBackupRoomsPath = [[storeBackupPath stringByAppendingPathComponent:backupEventStreamToken]
+                                stringByAppendingPathComponent:kMXFileStoreRoomsFolder];
+    }
+
+    return backupEventStreamToken;
+}
+
+- (void)checkFolderExistenceForRoom:(NSString*)roomId forBackup:(BOOL)backup
+{
+    NSString *roomFolder = [self folderForRoom:roomId forBackup:backup];
     if (![NSFileManager.defaultManager fileExistsAtPath:roomFolder])
     {
         [[NSFileManager defaultManager] createDirectoryAtPath:roomFolder withIntermediateDirectories:YES attributes:nil error:nil];
     }
 }
 
-- (NSString*)messagesFileForRoom:(NSString*)roomId
+- (NSString*)messagesFileForRoom:(NSString*)roomId forBackup:(BOOL)backup
+{
+    return [[self folderForRoom:roomId forBackup:backup] stringByAppendingPathComponent:kMXFileStoreRoomMessagesFile];
+}
+
+- (NSString*)stateFileForRoom:(NSString*)roomId forBackup:(BOOL)backup
 {
-    return [[storeRoomsPath stringByAppendingPathComponent:roomId] stringByAppendingPathComponent:kMXFileStoreRoomMessagesFile];
+    return [[self folderForRoom:roomId forBackup:backup] stringByAppendingPathComponent:kMXFileStoreRoomStateFile];
 }
 
-- (NSString*)stateFileForRoom:(NSString*)roomId
+- (NSString*)accountDataFileForRoom:(NSString*)roomId forBackup:(BOOL)backup
 {
-    return [[storeRoomsPath stringByAppendingPathComponent:roomId] stringByAppendingPathComponent:kMXFileStoreRoomStateFile];
+    return [[self folderForRoom:roomId forBackup:backup] stringByAppendingPathComponent:kMXFileStoreRoomAccountDataFile];
 }
 
-- (NSString*)accountDataFileForRoom:(NSString*)roomId
+- (NSString*)readReceiptsFileForRoom:(NSString*)roomId forBackup:(BOOL)backup
 {
-    return [[storeRoomsPath stringByAppendingPathComponent:roomId] stringByAppendingPathComponent:kMXFileStoreRoomAccountDataFile];
+    return [[self folderForRoom:roomId forBackup:backup] stringByAppendingPathComponent:kMXFileStoreRoomReadReceiptsFile];
 }
 
-- (NSString*)readReceiptsFileForRoom:(NSString*)roomId
+- (NSString*)metaDataFileForBackup:(BOOL)backup
 {
-    return [[storeRoomsPath stringByAppendingPathComponent:roomId] stringByAppendingPathComponent:kMXFileStoreRoomReadReceiptsFile];
+    if (!backup)
+    {
+        return [storePath stringByAppendingPathComponent:kMXFileStoreMedaDataFile];
+    }
+    else
+    {
+        return [[storeBackupPath stringByAppendingPathComponent:backupEventStreamToken] stringByAppendingPathComponent:kMXFileStoreMedaDataFile];
```

Comment:
It is not a big issue as the value returned by this method is used only if there is a file to backup. On the first login, there is not yet a MetaData file so no issue.